cpp/src/plasma/CMakeLists.txt

```diff
@@ -160,6 +160,41 @@ install(
   FILES "${CMAKE_CURRENT_BINARY_DIR}/plasma.pc"
   DESTINATION "${CMAKE_INSTALL_LIBDIR}/pkgconfig/")
 
+if(ARROW_PLASMA_JAVA_CLIENT)
+  # Plasma java client support
+  find_package(JNI REQUIRED)
+  # add jni support
+  include_directories(${JAVA_INCLUDE_PATH})
+  include_directories(${JAVA_INCLUDE_PATH2})
+  if (JNI_FOUND)
+    message (STATUS "JNI_INCLUDE_DIRS = ${JNI_INCLUDE_DIRS}")
+    message (STATUS "JNI_LIBRARIES = ${JNI_LIBRARIES}")
+  else()
+    message (WARNING "Could not find JNI")
+  endif()
+
+  add_compile_options("-I$ENV{JAVA_HOME}/include/")
+  if(WIN32)
+    add_compile_options("-I$ENV{JAVA_HOME}/include/win32")
+  elseif(APPLE)
+    add_compile_options("-I$ENV{JAVA_HOME}/include/darwin")
+  else() # linux
+    add_compile_options("-I$ENV{JAVA_HOME}/include/linux")
+  endif()
+
+  include_directories("${CMAKE_CURRENT_LIST_DIR}/lib/java")
+
+  file(GLOB PLASMA_LIBRARY_EXT_java_SRC
+    lib/java/*.cc lib/*.cc)
+  add_library(plasma_java SHARED
+    ${PLASMA_LIBRARY_EXT_java_SRC})
+
+  if(APPLE)
+    target_link_libraries(plasma_java plasma_static ${PLASMA_LINK_LIBS} "-undefined dynamic_lookup" -Wl,-force_load,${FLATBUFFERS_STATIC_LIB} ${FLATBUFFERS_STATIC_LIB} ${PTHREAD_LIBRARY})
```

Comment:
Why do we need the ` -Wl,-force_load,` part here?

Reply:
-Wl,-force_load for add runtime library path here